simulation/traffic_simulator/test/src/test_hdmap_utils.cpp

```diff
@@ -122,6 +122,116 @@ TEST(HdMapUtils, RoadShoulder)
   }
 }
 
+TEST(HdMapUtils, CanonicalizeNegative)
+{
+  std::string path =
+    ament_index_cpp::get_package_share_directory("traffic_simulator") + "/map/lanelet2_map.osm";
+  geographic_msgs::msg::GeoPoint origin;
+  origin.latitude = 35.61836750154;
+  origin.longitude = 139.78066608243;
+  hdmap_utils::HdMapUtils hdmap_utils(path, origin);
+
+  double canonicalized_lanelet_offset = -22;
+  const auto non_canonicalized_lanelet_pose =
+    traffic_simulator::helper::constructLaneletPose(34564, canonicalized_lanelet_offset, 0);
+  const auto canonicalized_lanelet_pose = std::get<std::optional<traffic_simulator::LaneletPose>>(
+    hdmap_utils.canonicalizeLaneletPose(non_canonicalized_lanelet_pose));
+
+  EXPECT_EQ(canonicalized_lanelet_pose.value().lanelet_id, 34576);
+  EXPECT_EQ(
+    canonicalized_lanelet_pose.value().s, canonicalized_lanelet_offset +
+                                            hdmap_utils.getLaneletLength(34570) +
+                                            hdmap_utils.getLaneletLength(34576));
+}
+
+TEST(HdMapUtils, CanonicalizePositive)
+{
```

Comment:
Please describe the test target and expected result in comment.

Reply:
Comment added